modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java

```diff
@@ -9115,6 +9116,178 @@ public void testPutCustomObjectEntryUnlinkNestedCustomObjectEntriesByExternalRef
+
+		ObjectDefinition objectDefinition =
+			ObjectDefinitionTestUtil.publishObjectDefinition(
+				true, objectDefinitionName,
+				Collections.singletonList(
+					new AttachmentObjectFieldBuilder(
+					).labelMap(
+						LocalizedMapUtil.getLocalizedMap("localizedAttachment")
+					).localized(
+						true
+					).name(
+						"localizedAttachment"
+					).objectFieldSettings(
+						Arrays.asList(
+							new ObjectFieldSettingBuilder(
+							).name(
+								ObjectFieldSettingConstants.
+									NAME_ACCEPTED_FILE_EXTENSIONS
+							).value(
+								"txt"
+							).build(),
+							new ObjectFieldSettingBuilder(
+							).name(
+								ObjectFieldSettingConstants.NAME_FILE_SOURCE
+							).value(
+								ObjectFieldSettingConstants.VALUE_USER_COMPUTER
+							).build(),
+							new ObjectFieldSettingBuilder(
+							).name(
+								ObjectFieldSettingConstants.NAME_MAX_FILE_SIZE
+							).value(
+								"100"
+							).build(),
+							new ObjectFieldSettingBuilder(
+							).name(
+								ObjectFieldSettingConstants.
+									NAME_SHOW_FILES_IN_DOCS_AND_MEDIA
+							).value(
+								StringPool.TRUE
+							).build(),
+							new ObjectFieldSettingBuilder(
+							).name(
+								ObjectFieldSettingConstants.
+									NAME_STORAGE_DL_FOLDER_PATH
+							).value(
+								StringPool.SLASH + objectDefinitionName
+							).build())
+					).build()),
+				ObjectDefinitionConstants.SCOPE_COMPANY,
+				TestPropsValues.getUserId());
+
+		JSONObject jsonObject = HTTPTestUtil.invokeToJSONObject(
+			String.valueOf(_jsonFactory.createJSONObject()),
+			objectDefinition.getRESTContextPath(), Http.Method.POST);
+
+		String fileContent1 = RandomTestUtil.randomString();
+
+		com.liferay.object.rest.dto.v1_0.FileEntry fileEntry1 = _toFileEntry(
+			Base64::encode, fileContent1,
+			RandomTestUtil.randomString() + ".txt", null, null);
+
+		fileEntry1.setExternalReferenceCode(RandomTestUtil.randomString());
+
+		Group group = GroupTestUtil.addGroup();
+
+		Scope scope = new Scope();
+
+		scope.setExternalReferenceCode(group.getExternalReferenceCode());
+		scope.setType(Scope.Type.SITE);
+
+		fileEntry1.setScope(scope);
+
+		String fileContent2 = RandomTestUtil.randomString();
+
+		com.liferay.object.rest.dto.v1_0.FileEntry fileEntry2 = _toFileEntry(
+			Base64::encode, fileContent2,
+			RandomTestUtil.randomString() + ".txt", null, null);
+
+		fileEntry2.setExternalReferenceCode(RandomTestUtil.randomString());
+		fileEntry2.setScope(scope);
+
+		String endpoint = StringBundler.concat(
+			objectDefinition.getRESTContextPath(), StringPool.SLASH,
+			jsonObject.get("id"));
+
+		jsonObject = HTTPTestUtil.invokeToJSONObject(
+			JSONUtil.put(
+				"localizedAttachment_i18n",
+				JSONUtil.put(
+					"en_US",
+					JSONFactoryUtil.createJSONObject(fileEntry1.toString())
+				).put(
+					"pt_BR",
+					JSONFactoryUtil.createJSONObject(fileEntry2.toString())
+				)
+			).toString(),
+			endpoint + "?nestedFields=localizedAttachment.fileBase64",
```

Comment:
I am going to discuss with our PO if this is the expected/desired behaviour for the base64 decoding process when the field is localized

Reply:
All right.